### PR TITLE
Add prometheus metrics for pipeline runner

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,33 @@ ds1_multiply [type=multiply times=100];
...
+Help: "How long each pipeline task took to execute",
+
+Name: "pipeline_task_http_fetch_time",
+Help: "Time taken to fully execute the HTTP request",
+
+Name: "pipeline_task_http_response_body_size",
+Help: "Size (in bytes) of the HTTP response body",
+
+Name: "pipeline_runs_queued",
+Help: "The total number of pipline runs that are awaiting execution",
```
Comment (connorwstein) [inline docs/CHANGELOG.md:45]: typo - "pipeline"

File: core/services/prom_reporter_test.go [hunk 2]
```diff
@@ -39,6 +44,33 @@ func Test_PromReporter_OnNewLongestChain(t *testing.T) {
...
+		reporter.OnNewLongestChain(context.Background(), head)
+
+		backend.AssertExpectations(t)
+	})
+
+	t.Run("with unfinished pipeline task runs", func(t *testing.T) {
+		store, cleanup := cltest.NewStore(t)
+		defer cleanup()
+
+		require.NoError(t, store.DB.Exec(`SET CONSTRAINTS pipeline_task_runs_pipeline_run_id_fkey DEFERRED`).Error)
```
Comment (connorwstein) [inline core/services/prom_reporter_test.go:60]: we don't need this anymore with the migration right?

File: core/services/pipeline/orm.go [hunk 3]
```diff
@@ -278,6 +310,16 @@ func (o *orm) processNextUnclaimedTaskRun(ctx context.Context, fn ProcessTaskRun
 			if err != nil {
 				return errors.Wrap(err, "could not notify pipeline_run_completed")
 			}
+
+			elapsed := time.Since(ptRun.CreatedAt)
+			promPipelineRunTotalTimeToCompletion.WithLabelValues(pipelineSpecIDStr).Set(float64(elapsed))
+			if finalErrors, is := result.Error.(FinalErrors); is && finalErrors.HasErrors() {
```
Comment (connorwstein) [inline core/services/pipeline/orm.go:316]: these are the same line `promPipelineRunErrors.WithLabelValues(pipelineSpecIDStr).Inc()` - do we just want to add an `|| result.Error != nil`?

**Review by connorwstein — DISMISSED** [review]

File: core/services/prom_reporter_test.go (hunk 2 above)
Comment (samsondav) [reply core/services/prom_reporter_test.go:60]: Yes we do. This is required in addition to the migration

**Review by connorwstein — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Pull request merged**